tidy3d/components/medium.py

```diff
@@ -5974,6 +5982,172 @@ def perturbed_copy(
         return CustomPoleResidue.parse_obj(new_dict)
 
 
+class SkinDepthFitterParam(Tidy3dBaseModel):
+    """Advanced parameters for fitting complex-valued skin depth ``2j/k`` of a :class:`.LossyMetalMedium`
+    over its frequency bandwidth, where k is the complex-valued wavenumber inside the lossy metal. Real part
+    of this quantity corresponds to the physical skin depth.
+    """
+
+    max_num_poles: pd.PositiveInt = pd.Field(
+        LOSSY_METAL_DEFAULT_MAX_POLES,
+        title="Maximal Number Of Poles",
+        description="Maximal number of poles in complex-conjugate pole residue model for "
+        "fitting complex-valued skin depth.",
+    )
+
+    tolerance_rms: pd.NonNegativeFloat = pd.Field(
+        LOSSY_METAL_DEFAULT_TOLERANCE_RMS,
+        title="Tolerance In Fitting",
+        description="Tolerance in fitting complex-valued skin depth.",
+    )
+
+    frequency_sampling_points: pd.PositiveInt = pd.Field(
+        LOSSY_METAL_DEFAULT_SAMPLING_FREQUENCY,
+        title="Number Of Sampling Frequencies",
+        description="Number of sampling frequencies used in fitting.",
+    )
+
+    log_sampling: bool = pd.Field(
+        True,
+        title="Frequencies Sampling In Log Scale",
+        description="Whether to sample frequencies logarithmically (``True``),  "
+        "or linearly (``False``).",
+    )
+
+
+class LossyMetalMedium(Medium):
+    """Lossy metal modeled with a surface impedance boundary condition (SIBC).
+
+    Notes
+    -----
+
+        SIBC is suitable when the skin depth is much smaller than the structure feature size.
+
+    Example
+    -------
+    >>> lossy_metal = LossyMetalMedium(conductivity=10, frequency_range=(9e9, 10e9))
+
+    """
+
+    permittivity: Literal[1] = pd.Field(
+        1.0, title="Permittivity", description="Relative permittivity.", units=PERMITTIVITY
+    )
+
+    frequency_range: FreqBound = pd.Field(
+        ...,
```

Comment:
I wonder if we could come up with a sensible default for frequency_range based on the sources in the simulation. For example, I do something similar for nonlinear materials -- freq0 is a parameter for the nonlinear model, but the default can often be automatically determined from the sources in the simulation. You could probably do something similar with fwidth.

https://github.com/flexcompute/tidy3d/blob/61966a923dc537b9576cfe888b9de5208d9005be/tidy3d/components/medium.py#L162

Reply:
That's a good idea. So we get `frequency_range` from source bandwidth if it's `None`, or use it directly if it's not `None`. One concern that I have is that one might be interested in visualizing the fitting result. If taken from the source, that seems only possible until simulation object has been initialized, or one has to manually enter the frequency range in the `plot` method. So not sure which approach is the best.